### PR TITLE
update bookmarkvalidation warning message

### PR DESCRIPTION
**Pull request by ansyral**
@vwxyzh @herohua  @superyyrrzz @hellosnow @qinezh @chenkennt

### CODE REVIEW

**Comment by qinezh** [comment]
:shipit:

**Comment by hellosnow** [comment]
:shipit:

**Comment by superyyrrzz** [comment]
Question: what if the link is resolved from cross reference?

File: src/Microsoft.DocAsCode.Build.Common/ValidateBookmark.cs [hunk 1]
```diff
@@ -46,7 +46,7 @@ public override void Handle(HtmlDocument document, ManifestItem manifestItem, st
                  let index = link.IndexOfAny(new[] { '?', '#' })
                  let decodedLink = HttpUtility.UrlDecode(link.Remove(index))
                  where !WhiteList.Contains(bookmark) && PathUtility.IsRelativePath(decodedLink)
-                 select new LinkItem { Title = node.InnerText, Href = TransformPath(outputFile, decodedLink), Bookmark = bookmark, SourceFile = node.GetAttributeValue("sourceFile", null), SourceLineNumber = node.GetAttributeValue("sourceStartLineNumber", 0), TargetLineNumber = node.Line }).ToList();
+                 select new LinkItem { Title = node.InnerText, Href = TransformPath(outputFile, decodedLink), Bookmark = bookmark, SourceFile = HttpUtility.UrlDecode(node.GetAttributeValue("sourceFile", null)), SourceLineNumber = node.GetAttributeValue("sourceStartLineNumber", 0), TargetLineNumber = node.Line }).ToList();
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Common/ValidateBookmark.cs]: HtmlDecode

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by ansyral** [comment]
need to keep src md info to tell []() or <xref:>

---
In reply to: [259580923](https://github.com/dotnet/docfx/pull/880#issuecomment-259580923) [](ancestors = 259580923)

**Pull request merged**